packages/core/src/model-manager.ts

```diff
@@ -1,40 +1,49 @@
 'use strict';
 
-const Toposort = require('toposort-class');
-const _ = require('lodash');
+import defaults from 'lodash/defaults';
+import Toposort from 'toposort-class';
+import type { ModelStatic } from './model';
+import type { Sequelize } from './sequelize';
 
 export class ModelManager {
-  constructor(sequelize) {
+  declare sequelize: Sequelize;
+  declare models: ModelStatic[];
+
+  constructor(sequelize: Sequelize) {
     this.models = [];
     this.sequelize = sequelize;
   }
 
-  addModel(model) {
+  addModel<T extends ModelStatic>(model: T): T {
     this.models.push(model);
     this.sequelize.models[model.name] = model;
 
     return model;
   }
 
-  removeModel(modelToRemove) {
-    this.models = this.models.filter(model => model.name !== modelToRemove.name);
+  removeModel(modelToRemove: ModelStatic): void {
+    this.models = this.models.filter(
+      model => model.name !== modelToRemove.name,
+    );
 
     delete this.sequelize.models[modelToRemove.name];
   }
 
-  getModel(modelName) {
+  getModel(modelName: string): ModelStatic | undefined {
```

Comment:
Would it be better to be able to pass a type for the model static to the function to return a specific rather than generic model static instance?

```suggestion
  getModel<T extends ModelStatic>(modelName: string): T | undefined {
```

Reply:
it would be equivalent to casting, as we can't verify that it really is that type. In those instances, I personally prefer casting explicitly when needed rather than using a generic

Reply:
Thats fine, I'm happy with using casting.